routes/accounts.js

```diff
@@ -80,10 +81,22 @@ function accountRoutes (server, options, next) {
     handler: function (request, reply) {
       var sessionId = toBearerToken(request)
 
-      return accounts.findAll({
-        db: options.db,
-        bearerToken: sessionId,
-        include: request.query.include
+      admins.validateSession(sessionId)
+
+      .catch(function (error) {
+        if (error.name === dbErrors.MISSING_DOC.name) {
```

Comment:
sounds good to me on both points... I'll finish this PR and set the tag to ready. Then start another for cleaning up the error checking in other modules.